apps/src/lib/node/ledger/shell/prepare_proposal.rs

```diff
@@ -42,8 +42,18 @@ mod prepare_block {
                 // TODO: add some info logging
 
                 // add ethereum events as protocol txs
-                let mut txs =
-                    self.build_vote_extensions_txs(req.local_last_commit);
+                let mut txs = match req.local_last_commit {
+                    Some(local_last_commit) => {
+                        self.build_vote_extensions_txs(local_last_commit)
+                    }
+                    None => {
+                        tracing::info!(
+                            "No local last commit, skipping including vote \
+                             extensions"
```

Comment:
can you somehow fit "Ethereum events" into the log message? so that we can distinguish these vote extensions from the validator set update log messages.

Reply:
In this case, we don't get a vote extension at all (so neither Ethereum events nor validator set updates). I'm not sure when or if this code path can happen - in the [protobuf](https://github.com/tendermint/tendermint/blob/a1104b98d2d2f04855d59af98036a2278aab23d2/proto/tendermint/abci/types.proto#L131) the `local_last_commit` field is marked as not nullable and the generated Go struct has it as a non-optional field. For the type we are using from `tendermint-rs`, this field is an `Option<ExtendedCommitInfo>`. Need to look into whether the types are out of sync somehow

Reply:
Re the error in https://github.com/anoma/namada/issues/251 , `local_last_commit` was `Some(_)` but then the single `ExtendedVoteInfo` inside it had an empty `vote_extension` field. So this code path wasn't being hit in the end